src/Microsoft.AspNet.OData.Shared/TypelessPatchMethodHandler.cs

```diff
@@ -0,0 +1,196 @@
+    public abstract class TypelessPatchMethodHandler 
+    {
+        /// <summary>
+        /// TryCreate method to create a new object.
+        /// </summary>
+        /// <param name="changedObj">Changed object which can be appied on creted object, optional</param>
+        /// <param name="createdObject">The created object (Typeless)</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns></returns>
+        public abstract PatchStatus TryCreate(IEdmChangedObject changedObj, out EdmStructuredObject createdObject, out string errorMessage);
+
+        /// <summary>
+        /// TryGet method to which pointer to TryGet method can be assigned to.  This tries to Get based on a keyvalues.
+        /// </summary>
+        /// <param name="keyValues">Key value pair for the entity keys</param>        
+        /// <param name="originalObject">Object to return</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns></returns>
+        public abstract PatchStatus TryGet(IDictionary<string, object> keyValues, out EdmStructuredObject originalObject, out string errorMessage);
+
+        /// <summary>
+        ///  TryDelete delegate to which the pointer to TryDelete Method can be assigned to, which will delete the object based on keyvalue pairs
+        /// </summary>
+        /// <param name="keyValues"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns></returns>
+        public abstract PatchStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage);
+
+        /// <summary>
+        /// Get the PatchHandler for the nested type
+        /// </summary>
+        /// <param name="parent">Parent instance.</param>
+        /// <param name="navigationPropertyName">The name of the navigation property for the handler</param>
+        /// <returns></returns>
+        public abstract TypelessPatchMethodHandler GetNestedPatchHandler(EdmStructuredObject parent, string navigationPropertyName);
+    }
+
+       
+    internal class DefaultTypelessPatchHandler : TypelessPatchMethodHandler
+    {
+        IEdmEntityType entityType;
+        ICollection<EdmStructuredObject> originalList;
+
+        public DefaultTypelessPatchHandler(ICollection<EdmStructuredObject> originalList, IEdmEntityType entityType)
+        {
+            this.entityType = entityType;
+            this.originalList = originalList?? new List<EdmStructuredObject>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override PatchStatus TryGet(IDictionary<string, object> keyValues, out EdmStructuredObject originalObject, out string errorMessage)
+        {
+            PatchStatus status = PatchStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = default(EdmStructuredObject);
```

Comment:
>originalObject [](start = 12, length = 14)

move this to your exception handling so you don't needlessly create a defult instance of the EdmStructuredObject in the success case.